src/test/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncherIntegrationIT.java

```diff
@@ -16,235 +16,122 @@
+		String taskName = randomName();
+		AppDefinition definition = new AppDefinition(taskName, null);
 		Resource resource = testApplication();
-		AppDeploymentRequest request = new AppDeploymentRequest(definition, resource,
-				Collections.singletonMap("spring.cloud.deployer.kubernetes.deploymentLabels", "label1:value1,label2:value2"));
-
-		log.info("Launching {}...", request.getDefinition().getName());
-
-		String launchId = kubernetesTaskLauncher.launch(request);
-		Timeout timeout = deploymentTimeout();
-
-		await().pollInterval(Duration.ofMillis(timeout.pause))
-                .atMost(Duration.ofMillis(timeout.maxAttempts * timeout.pause))
-                .untilAsserted(() -> {
-			assertThat(taskLauncher().status(launchId).getState()).isEqualTo(LaunchState.running);
-        });
-
-		String taskName = request.getDefinition().getName();
+		AppDeploymentRequest request = new AppDeploymentRequest(definition, resource, deploymentProps);
 
-		log.info("Checking job pod spec labels of {}...", taskName);
-
-		List<Pod> pods = kubernetesClient.pods().withLabel("task-name", taskName).list().getItems();
+		log.info("Launching {}...", taskName);
+		String launchId = taskLauncher().launch(request);
+		awaitWithPollAndTimeout(deploymentTimeout())
+				.untilAsserted(() -> assertThat(taskLauncher().status(launchId).getState()).isEqualTo(LaunchState.running));
 
+		log.info("Checking task Pod for {}...", taskName);
+		List<Pod> pods = getPodsForTask(taskName); //kubernetesClient.pods().withLabel("task-name", taskName).list().getItems();
 		assertThat(pods).hasSize(1);
-
-		Pod pod = pods.get(0);
-
-		assertThat(pod.getSpec().getContainers().get(0).getPorts()).isEmpty();
-
-		Map<String, String> labels = pod.getMetadata().getLabels();
-
-		assertThat(labels).contains(entry("label1", "value1"), entry("label2", "value2"));
+		assertThat(pods).singleElement().satisfies(assertingConsumer);
 
 		log.info("Destroying {}...", taskName);
-
-		timeout = undeploymentTimeout();
-		kubernetesTaskLauncher.destroy(taskName);
-
-		await().pollInterval(Duration.ofMillis(timeout.pause))
-                .atMost(Duration.ofMillis(timeout.maxAttempts * timeout.pause))
-                .untilAsserted(() -> {
-			assertThat(taskLauncher().status(launchId).getState()).isEqualTo(LaunchState.unknown);
-        });
+		taskLauncher().destroy(taskName);
+		awaitWithPollAndTimeout(undeploymentTimeout())
+				.untilAsserted(() -> assertThat(taskLauncher().status(launchId).getState()).isEqualTo(LaunchState.unknown));
 	}
 
 	@Test
-	public void testTaskAdditionalContainer() {
-		log.info("Testing {}...", "TaskAdditionalContainer");
-
-		KubernetesTaskLauncher kubernetesTaskLauncher = new KubernetesTaskLauncher(new KubernetesDeployerProperties(),
-				new KubernetesTaskLauncherProperties(), kubernetesClient);
-
+	void cleanupDeletesTaskPod(TestInfo testInfo) {
```

Comment:
And this one test was the whole motivation for this code proposal. It varies little w/ the main flow of the others besides that fact that it calls `cleanup` directly, rather than `destroy`. It then, validates that it can't find the pods after the delete - which seems to be the best we can do to verify that the pod was actually deleted. 

And once I got to this point I realized we could have easily mocked the PodList/JobList code out and done this w/ a simple unit test. All we really want to know is that "when I call cleanup(taskId) that the underlying task job/podList has `delete` called on it as well. Of course the downside of that approach is the mocking required which drives us pretty deep down into a specific impl of api calls to get back the pod/job list etc.. I am inclined to leave this verification here in the IT based on this fact of the unit test mock approach being more brittle. However, I am open to switching to unit test path if others feel strongly in favor of the unit test approach.